libs/partners/mistralai/langchain_mistralai/embeddings.py

```diff
@@ -209,16 +211,27 @@ def embed_documents(self, texts: List[str]) -> List[List[float]]:
             List of embeddings, one for each text.
         """
         try:
-            batch_responses = (
-                self.client.post(
+            batch_responses = []
+
+            @retry(
+                retry=retry_if_exception_type(Exception),
+                wait=wait_fixed(30),  # Wait 30 seconds between retries
+                stop=stop_after_attempt(5),  # Stop after 5 attempts
+            )
+            def _embed_batch(batch: List[str]) -> Response:
+                response = self.client.post(
                     url="/embeddings",
                     json=dict(
                         model=self.model,
                         input=batch,
                     ),
                 )
-                for batch in self._get_batches(texts)
-            )
+                if response.status_code == 429:
+                    raise Exception("Requests rate limit exceeded")
```

Comment:
This code takes an exception that was good and informative and turns it into one that's a broad Exception of type Exception -- this is usually not a good pattern for exception handling. Stack trace will be partially lost, the exception type is less specific etc.

Reply:
- Their were no specific exception being raised to being with. But I can change it from a general Exception.